app/Auth0/Auth0ClientsForIntegration.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Auth0;
+
+use InvalidArgumentException;
+use Ramsey\Uuid\UuidInterface;
+
+final class Auth0ClientsForIntegration
+{
+    private array $auth0ClientPerTenant = [];
+
+    public function __construct(
+        public readonly UuidInterface $integrationId,
```

Comment:
Would it be better to remove this redundant parameter `$integrationId` and make sure that all passed `Auth0Client` have the same integration id?

Reply:
I initially did it like that but that doesn't work when an integration has 0 clients, for example when all tenant config is disabled (in tests or when working locally without tenant config for some reason) or when handling the `CreateClients` async. Because then this object will have no `$integrationId` at all

Reply:
Does that also mean that the passed-in array of `$auth0Clients` is empty?

Reply:
Yes, for example if you call `Auth0ClientRepository::getByIntegrationId()` and get 0 rows because the async command is not handled yet or there are no configured tenants, then it will return an instance of `Auth0ClientsForIntegration` with the given `$integrationId` and an array of 0 values for the clients. Since there are 0 rows/clients at that point, `Auth0ClientsForIntegration` cannot get the `$integrationId` from the client objects themselves so it needs to be injected in the constructor.

Reply:
I'm not sure I understand. If there are 0 rows the array of `$auth0Clients` will be empty, so the foreach won't do anything and the validation won't happen.

Reply:
Maybe there is some confusion because `$integrationId` is not just an argument but also a property? It's just set automatically now using the `public readonly`?

Reply:
So then I think that `$integrationId` is not needed as a parameter?

Reply:
It's not just a parameter but a property (our messages probably crossed). I see now that it's unused out of the class though so I could remove it. But then I would rename the class to just `Auth0Clients` since the integration context is lost then

Reply:
Keep it like it is, it's fine.

Reply:
Update 2: I'm going to remove this class completely and just work with `array` 😄

Reply:
Maybe a Laravel `Collection` is a good alternative?

Reply:
At first glance that is more of a "util" class that provides chainable methods like `map()` etc so you don't need to use `array_map()`, and you can easily create it from an `array` when needed. So my instinct would be to just work with `array` to keep it a bit less coupled and use `collect()` or `new Collection()` where you want the helper methods. Or am I missing something like type checking? https://laravel.com/docs/9.x/collections

Reply:
I've just refactored it to `array` now in [993c7bd](https://github.com/cultuurnet/publiq-platform/pull/28/commits/993c7bd1c049109125c07d421fa5a94b5a50e40b) and [0576d60](https://github.com/cultuurnet/publiq-platform/pull/28/commits/0576d60492fa4a44face023a16817a0f9c59d899) but if `Collection` is in fact the way to go it's easy to change.

Reply:
I tried to return a `Collection` from the repositories. Make it also easy to work with in the HTTP Controller layer or even the Inertia views.